src/Native/System.Net.Security.Native/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+project(System.Net.Security.Native)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+add_definitions(-DPIC=1)
+
+if (HAVE_HEIMNTLM_HEADERS)
+   find_library(LIB_HEIMNTLM NAMES heimntlm PATHS /usr/heimdal/lib)
+   include_directories(/usr/heimdal/include)
+
+   if(LIB_HEIMNTLM STREQUAL LIB_HEIMNTLM-NOTFOUND)
+       message(FATAL_ERROR "Cannot find libheimntlm.so and System.Net.Security.Native cannot build without it. Try installing libheimntlm (or the appropriate package for your platform)")
+   endif()
+
+
+   set(NATIVENTLM_SOURCES
+     pal_ntlmapi.cpp
+   )
+
+   add_library(System.Net.Security.Native
+    SHARED
+    ${NATIVENTLM_SOURCES}
+   )
+
+   target_link_libraries(System.Net.Security.Native
+           ${LIB_HEIMNTLM}
+           ${OPENSSL_CRYPTO_LIBRARY}
+   )
+
+   install (TARGETS System.Net.Security.Native DESTINATION .)
+
+elseif (CMAKE_SYSTEM_NAME STREQUAL Linux)
+   message (FATAL_ERROR "cannot find libheim ntlm headers and System.Net.Security.Native cannot build without it. Try installing libheimntlm (or the appropriate package for your platform)")
+else ()
```

Comment:
I find this construction odd.

If we already detected as having the headers/library then use them (failing if we can't find the library).  If we didn't find the headers and we're Linux, fail.  Otherwise write a message and don't build anything.

I think right now we have OS X and Linux expected to succeed.  What about FreeBSD?  If someone were to come along and start porting to Solaris, what state would they be in?  Is the managed code guarding against this, too?  On those platforms do we expect to have no System.Net.Security.Native library at all, or should it have one that at startup instead indicates that there's no NTLM support?